src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -2824,10 +2796,10 @@ err_t sockinfo_tcp::accept_lwip_cb(void *arg, struct tcp_pcb *child_pcb, err_t e
 
 	conn->unlock_tcp_con();
 
-	new_sock->lock_tcp_con();
-#ifdef DEFINED_SOCKETXTREME
+	/* Do this after auto_accept_connection() call */
 	new_sock->m_parent = NULL;
-#endif // DEFINED_SOCKETXTREME	
```

Comment:
Why not under lock ? I do not think it matters but if it does not matter why you change the order?